core/src/main/java/com/linecorp/armeria/internal/DefaultTimeoutController.java

```diff
@@ -35,16 +36,23 @@
  */
 public class DefaultTimeoutController implements TimeoutController {
 
+    enum State {
+        DISABLED,
+        SCHEDULED,
+        TIMED_OUT,
+    }
+
     @Nullable
     private TimeoutTask timeoutTask;
     private final EventLoop eventLoop;
 
     private long timeoutMillis;
-    private long firstStartTimeNanos;
-    private long lastStartTimeNanos;
+    private long firstExecutionTimeNanos;
+    private long lastExecutionTimeNanos;
```

Comment:
`0` has a meaning in nano time. Need to introduce a boolean?

Reply:
Ahh... I forget it. 😱